src/networks/mod.rs

```diff
@@ -176,7 +176,7 @@ impl ChainConfig {
         use mainnet::*;
         Self {
             network: NetworkChain::Mainnet,
-            genesis_cid: Some(GENESIS_CID.to_owned()),
```

Comment:
How about we go ahead and make the `genesis_cid` field in the `ChainConfig` an `Option<Cid>`? The more strongly typed the codebase is, the better.

Reply:
I'm okay with doing this in a separate PR to keep things small. Plus, the change may sprawl into several files.

Reply:
Issue #3247